src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -199,19 +200,55 @@ public bool IsConnected
             }
         }
 
+        public static IPEndPoint GetHandshakedEndPoint(INetworkPeer peer)
+        {
+            if (peer.Inbound && peer.State == NetworkPeerState.HandShaked)
```

Comment:
Note that the previous code checked the `Connected` state as well.

Reply:
The initial state for an inbound peer is `Connected`. Once we get the `VersionPayload` from the peer the state is set to `Handshaked`. I expect that `AddressFrom` would only be usable/valid once handshake has been completed as it arrives as part of the handshake payload.